### PR TITLE
use the vm service protocol's getMemoryUsage() API

### PR DESCRIPTION
**Pull request by devoncarew**
- switch to using the vm service protocol's `getMemoryUsage()` API (instead of polling isolates)
- address one related todo

cc @terrylucas

### CODE REVIEW

File: packages/devtools_shared/lib/src/memory/heap_sample.dart [hunk 1]
```diff
@@ -31,7 +29,6 @@ class HeapSample {
 
   Map<String, dynamic> toJson() => <String, dynamic>{
         'timestamp': timestamp,
-        'rss': rss,
```
Comment (terrylucas) [inline packages/devtools_shared/lib/src/memory/heap_sample.dart:34]: rss should be stored to both the JSON files (that we export) and for reloading too.

Comment (devoncarew) [reply packages/devtools_shared/lib/src/memory/heap_sample.dart:34]: restored

**Comment by devoncarew** [comment]
@terrylucas, can you re-review?

**Comment by devoncarew** [comment]
Landing, and can make additional revisions as necessary.

**Pull request merged**